mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala

```diff
@@ -640,4 +689,96 @@ private object RandomForestSuite {
     val (indices, values) = map.toSeq.sortBy(_._1).unzip
     Vectors.sparse(size, indices.toArray, values.toArray)
   }
+
+  /** Generate a label. */
+  private def generateLabel(rnd: Random, numClasses: Int): Double = {
+    rnd.nextInt(numClasses)
+  }
+
+  /** Generate a numeric value in the range [numericMin, numericMax]. */
+  private def generateNumericValue(rnd: Random, numericMin: Double, numericMax: Double) : Double = {
+    rnd.nextDouble() * (Math.abs(numericMax) + Math.abs(numericMin)) + numericMin
```

Comment:
Just: `rnd.nextInt(numericMax - numericMin + 1) + numericMin`

Reply:
Isn't there an issue being numericMax and numericMin double values? 
I have changed it to: rnd.nextDouble() * (numericMax - numericMin) + numericMin
I think it is in line with what you proposed (modulo the "+ 1" that I don't understand).

Reply:
Oh right, disregard that. I had in mind that they were ints.